resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -216,8 +216,70 @@ private[spark] class Client(
                 .endContainer()
               .endSpec()
             .done()
-          submitCompletedFuture.get(30, TimeUnit.SECONDS)
-        }
+          try {
+            submitCompletedFuture.get(30, TimeUnit.SECONDS)
```

Comment:
It seems like we consider submission to be complete only if the driver pod has entered the running state. It may make sense to instead report the submission as being complete once the APIServer has accepted the request, even if the pod is in the `Pending` state. Would that make sense?